### ISSUE TITLE
[Inductor] Inference Fuse Gelu 

### ISSUE THREAD

**Issue by eellison** [pytorch/pytorch#103480]
### 🚀 The feature, motivation and pitch

When you run :

```
TORCH_COMPILE_DEBUG=1 python benchmarks/dynamo/torchbench.py --device cuda --performance --backend inductor --float16 --inference --only hf_Bart 
```

You'll see that there are 13 GELU kernels which directly follow addmms and have nothing else to fuse to. For this model GELUs are about ~5% of inference cost. 

We do some amount of epilogue fusion, but it depends on MAX_AUTOTUNE_GEMM being set and for us to find a suitable triton mm replacement for addmm, which often fails.

Instead of relying on triton, we can reuse the existing GELU/ReLU activation support which is already exists in CuBlas and is bound in PyTorch: https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/cuda/Blas.cpp#L119-L123. 

When I measured this manually on one of the matmuls from this model, there was a 5% speedup from baseline, reducing the cost of GELU by 20%. This should be about a ~1% speedup for the model as a whole. 
 
 We'll want to register a graph rewriting pattern like https://github.com/pytorch/pytorch/blob/main/torch/_inductor/fx_passes/pad_mm.py which calls into `_addmm_activation` when `MAX_AUTOTUNE_GEMM=False`, and find a way for this to make its way into https://github.com/pytorch/pytorch/blob/7360d0f904bcf2ef32e05f3abf403d9ff4503234/torch/_inductor/kernel/mm.py#L141 when `MAX_AUTOTUNE_GEMM=True`.

Note: this is being written up as ramp up for team members, if you are an oss person looking to contribute, please take a look at https://github.com/pytorch/pytorch/issues/102207.

cc @ezyang @msaroufim @wconstab @bdhirsh @anijain2305 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @aakhundov @chenyang78

**Comment by aakhundov** [comment]
Submitted #103811 as a prerequisite to this PR: it enables the `addmm` + GELU epilogue fusion in `ATen/native/cuda/Blas.cpp` (currently [disabled](https://github.com/pytorch/pytorch/blob/15eed5b73ef1ebe0d1142d70bab7c20300a2aa2c/aten/src/ATen/native/cuda/Blas.cpp#L293-L296) in the trunk).

**Comment by OrenLeung** [comment]
@ellison would a similar perf increase be possible with conv + activation fusion too?

**Comment by eellison** [comment]
Yea, definitely. One difficulty though is the CUDNN api causes a pretty bad regression until CUDNN 8.8.1: https://gist.github.com/eellison/73d1b512845ec07f03f703a625e028c8. Maybe before then, I'm not sure (cc @eqy).

Work is ongoing for a cutlass integration that would be alternate way to fuse conv-relu, although that would only fire in MAX_AUTOTUNE mode. 

For 1x1 convs with MAX_AUTOTUNE we will already fuse convs, see https://github.com/pytorch/pytorch/pull/107004. That may requiring freezing to be on to hit for many models in order to fold batchnorm TORCHINDUCTOR_FREEZING=1. That api is not fully cleaned up and published yet.
